tests/unit/python/foglamp/services/core/api/test_scheduler_api.py

```diff
@@ -0,0 +1,443 @@
+            assert 404 == resp.status
+            assert 'No such Schedule: {}.'.format(self._random_uuid) == resp.reason
+        patch_get_schedule.assert_called_once_with(uuid.UUID('{}'.format(self._random_uuid)))
+
+    @pytest.mark.parametrize("request_data, response_code, error_message, storage_return", [
+        ({"type": 'bla'}, 400, "Error in type: bla", {'rows': [{'name': 'bla'}], 'count': 1}),
+        ({"day": 'bla'}, 400, "Error in day: bla", {'rows': [{'name': 'bla'}], 'count': 1}),
+        ({"time": 'bla'}, 400, "Error in time: bla", {'rows': [{'name': 'bla'}], 'count': 1}),
+        ({"repeat": 'bla'}, 400, "Error in repeat: bla", {'rows': [{'name': 'bla'}], 'count': 1}),
+        ({"type": 2, "name": "sch1", "process_name": "p1"}, 404,
+         "Errors in request: Schedule day and time cannot be empty for TIMED schedule.",
+         {'rows': [{'name': 'bla'}], 'count': 1}),
+        ({"type": 2, "day": 9, "time": 1, "name": "sch1", "process_name": "p1"}, 404,
+         "Errors in request: Day must be an integer and in range 1-7.",
+         {'rows': [{'name': 'bla'}], 'count': 1}),
+        ({"type": 2, "day": 5, "time": -1, "name": "sch1", "process_name": "p1"}, 404,
+         "Errors in request: Time must be an integer and in range 0-86399.",
+         {'rows': [{'name': 'bla'}], 'count': 1}),
+        ({"type": 200}, 404,
+         "Errors in request: Schedule type error: 200",
+         {'rows': [{'name': 'bla'}], 'count': 1}),
+        ({"type": 1, "name": "sch1", "process_name": "p1"}, 404,
+         "Errors in request: No such Scheduled Process name: p1",
+         {'rows': [], 'count': 0}),
+    ])
+    async def test_update_schedule_bad_data(self, client, request_data, response_code, error_message, storage_return):
+        async def mock_coro():
+            schedule = StartUpSchedule()
+            schedule.schedule_id = self._random_uuid
+            schedule.exclusive = True
+            schedule.enabled = True
+            schedule.name = "foo"
+            schedule.process_name = "bar"
+            schedule.repeat = timedelta(seconds=30)
+            schedule.time = None
+            schedule.day = None
+            return schedule
+
+        storage_client_mock = MagicMock(StorageClient)
+        response = storage_return
+        with patch.object(connect, 'get_storage', return_value=storage_client_mock):
+            with patch.object(storage_client_mock, 'query_tbl_with_payload', return_value=response):
+                with patch.object(server.Server.scheduler, 'get_schedule',
+                                  return_value=mock_coro()) as patch_get_schedule:
+                    resp = await client.put('/foglamp/schedule/{}'.format(self._random_uuid), data=json.dumps(request_data))
+                    assert response_code == resp.status
+                    assert error_message == resp.reason
+
+    async def test_delete_schedule(self, client):
```

Comment:
add 1 more test `test_delete_enabled_schedule` it should return something like 409 (enabled schedule can not be deleted)